src/sentry/static/sentry/app/views/dashboardsV2/widgetQueries.tsx

```diff
@@ -0,0 +1,173 @@
+import React from 'react';
+import isEqual from 'lodash/isEqual';
+
+import {doEventsRequest} from 'app/actionCreators/events';
+import {Client} from 'app/api';
+import {
+  getDiffInMinutes,
+  getInterval,
+  isMultiSeriesStats,
+} from 'app/components/charts/utils';
+import {
+  EventsStats,
+  GlobalSelection,
+  MultiSeriesEventsStats,
+  Organization,
+} from 'app/types';
+import {Series} from 'app/types/echarts';
+import {parsePeriodToHours} from 'app/utils/dates';
+
+import {Widget} from './types';
+
+// Don't fetch more than 2000 bin as we're plotting on fewer pixels.
+const MAX_BIN_COUNT = 2000;
+
+function getWidgetInterval(
+  desired: string,
+  datetimeObj: Partial<GlobalSelection['datetime']>
+): string {
+  const selectedRange = getDiffInMinutes(datetimeObj);
+  const desiredPeriod = parsePeriodToHours(desired) * 60;
+  if (selectedRange / desiredPeriod > MAX_BIN_COUNT) {
```

Comment:
is it possible that desiredPeriod === 0 ?

Reply:
No, but it can be theoretically be -1, but that also shouldn't happen unless we make a breaking change to the interval size. Based on discussion in slack today, we'll be removing the interval property in the short term as a way to simplify widget creation.